src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh

```diff
@@ -8,6 +8,12 @@ if [[ -n "$VSCODE_SHELL_INTEGRATION" ]]; then
 	builtin return
 fi
 
+__vsc_dbg_trap="$(trap -p DEBUG)"
+if [[ "$__vsc_dbg_trap" =~ .*\[\[.* ]]; then
+	#https://github.com/microsoft/vscode/issues/157851
+	builtin return;
+fi
```

Comment:
This needs to happen next to the `if [ -z "$VSCODE_SHELL_INTEGRATION" ]; then` statement so regular init scripts run when injected. A comment above the if and I think we should inline the trap command so `__vsc_debug_trap` doesn't get set: `if [[ "$(trap -p DEBUG)" =~ .*\[\[.* ]]; then`